farmbot_ext/lib/farmbot_ext/amqp/connection_worker.ex

```diff
@@ -4,24 +4,29 @@ defmodule FarmbotExt.AMQP.ConnectionWorker do
   """
 
   use GenServer
-  alias FarmbotCore.Project
-  alias FarmbotExt.JWT
   require Logger
+  alias FarmbotExt.JWT
+  alias FarmbotCore.Project
+
+  @me __MODULE__
+
+  # IMPORTANT- prevents bad builds!:
+  FarmbotExt.fetch_impl!(@me, :network_impl)
 
   defstruct [:opts, :conn]
 
   def start_link(args) do
-    GenServer.start_link(__MODULE__, args, name: __MODULE__)
+    GenServer.start_link(@me, args, name: @me)
```

Comment:
this should probably be ```elixir
def start_link(args, opts \\ [name: __MODULE__]) do
   GenServer.start_link(__MODULE__, args, opts)
end
```
This will make for better testability later.